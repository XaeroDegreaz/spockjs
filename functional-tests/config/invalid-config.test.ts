import { transform } from '@babel/core';
import plugin from '@spockjs/babel-plugin-spock';

test('throws if the config contains an unknown key', () => {
  expect(() =>
    transform(`expect: 1 === 1;`, {
      plugins: [[plugin, { asdf: true }]],
    }),
  ).toThrowErrorMatchingSnapshot();
});

test('throws if the config contains an invalid data type', () => {
  expect(() =>
    transform(`expect: 1 === 1;`, {
      plugins: [[plugin, { powerAssert: 42 }]],
    }),
  ).toThrowErrorMatchingSnapshot();
});
