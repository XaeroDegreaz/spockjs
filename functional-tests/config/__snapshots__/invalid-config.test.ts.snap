// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throws if the config contains an invalid data type 1`] = `
"Invalid plugin configuration:
Invalid value 42 supplied to : Config/powerAssert: (boolean | undefined)/0: boolean
Invalid value 42 supplied to : Config/powerAssert: (boolean | undefined)/1: undefined"
`;

exports[`throws if the config contains an unknown key 1`] = `
"Invalid plugin configuration:
Invalid value true supplied to : Config/asdf: never"
`;
